### PR TITLE
Download the merged Cobertura report in the current build

### PR DESCRIPTION
**Pull request by jchen-parasoft**

### CODE REVIEW

File: PublishParasoftResults/BuildApiClient.ts [hunk 1]
```diff
@@ -71,6 +73,29 @@ export class BuildAPIClient {
         return [];
     }
 
+    async getCoberturaArtifactOfBuildById(buildId: number): Promise<BuildArtifact> {
+        return (await this.buildApi).getArtifact(this.projectName, buildId, COBERTURA_ARTIFACT_NAME);
+    }
+
+    async getCoberturaReportsOfArtifact(artifact: BuildArtifact): Promise<FileEntry[]> {
```
Comment (yshi-parasoft) [inline PublishParasoftResults/BuildApiClient.ts:80]: Please refactor this method and `getSarifReportsOfArtifact` to reduce duplicated code.

File: PublishParasoftResults/ParaReportPublishService.ts [hunk 2]
```diff
@@ -71,6 +71,12 @@ interface ReferenceBuildResult {
         buildNumber:  string | undefined,
         warningMessage: string | undefined
     }
+    codeCoverage?: {
```
Comment (yshi-parasoft) [inline PublishParasoftResults/ParaReportPublishService.ts:74]: What is the purpose of saving this information?

File: PublishParasoftResults/CoverageReportService.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    buildClient: BuildAPIClient;
+
+    constructor() {
+        this.buildClient = new BuildAPIClient();
+    }
+
+    async getMergedCoberturaReportByBuildId(buildId: number): Promise<FileEntry | undefined> {
+        const coberturaReports = await this.buildClient.getCoberturaReportsByBuildId(buildId);
+        const mergedCoberturaReports = coberturaReports.filter(coberturaReport => {
+            return coberturaReport.name == "CoberturaContainer/parasoft-merged-cobertura.xml";
```
Comment (rwang-parasoft) [inline PublishParasoftResults/CoverageReportService.ts:28]: We can simplify the code
```
return coberturaReports.find(coberturaReport => {
    return coberturaReport.name === "CoberturaContainer/parasoft-merged-cobertura.xml";
});
```

File: PublishParasoftResults/BuildApiClient.ts [hunk 4]
```diff
@@ -56,13 +58,22 @@ export class BuildAPIClient {
     }
 
     async getSarifReportsOfArtifact(artifact: BuildArtifact): Promise<FileEntry[]> {
+        return await this.getReportsOfArtifact(artifact, SARIF_FILE_SUFFIX);
+    }
+
+    async getCoberturaReportsByBuildId(buildId: number): Promise<FileEntry[]> {
+        const artifact = await (await this.buildApi).getArtifact(this.projectName, buildId, COBERTURA_ARTIFACT_NAME);
```
Comment (rwang-parasoft) [inline PublishParasoftResults/BuildApiClient.ts:65]: Check the artifact if it is null or undefined first, then get reports if not.

**Comment by yshi-parasoft** [comment]
From the description of the [task](https://jira.parasoft.com/browse/CICD-726), it appears that the action to "Download" should be included in this pull request. However, I did not see that the new service was introduced or that the method was used.

**Comment by yshi-parasoft** [comment]
> From the description of the [task](https://jira.parasoft.com/browse/CICD-726), it appears that the action to "Download" should be included in this pull request. However, I did not see that the new service was introduced or that the method was used.

OK, I see that this part will be done in [CICD-727](https://jira.parasoft.com/browse/CICD-727) according to the description. I'll merge this pull request and change the description of the current task instead.

**Pull request merged**